src/mdns/mdns_mdnssd.cpp

```diff
@@ -284,23 +284,23 @@ void PublisherMDnsSd::Update(MainloopContext &aMainloop)
         aMainloop.mMaxFd = std::max(aMainloop.mMaxFd, fd);
     }
 
-    for (Subscription &subscription : mSubscribedServices)
+    for (auto &subscription : mSubscribedServices)
```

Comment:
nit: here and everywhere below, you may use `const auto &`.

Reply:
I am doing that in #1122

I prefer to keep this PR focus on the memaccess fixing